### ISSUE TITLE
Crash in GcmIntentService

### ISSUE THREAD

**Issue by kistitan** [OneSignal/OneSignal-Android-SDK#195]
```Java
Fatal Exception: java.lang.NullPointerException
       at com.onesignal.NotificationBundleProcessor.ProcessFromGCMIntentService(NotificationBundleProcessor.java:53)
       at com.onesignal.GcmIntentService.onHandleIntent(GcmIntentService.java:58)
       at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:65)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:137)
       at android.os.HandlerThread.run(HandlerThread.java:60)
```

**Comment by jkasten2** [comment]
@kistitan Can you confirm which version of the SDK you seen this issue on? Please also include the Android version and device model if you can.

We call `setIntentRedelivery(true);` from our `GcmIntentService` which should resend the `Bundle` again if the service was restarted due to can out of memory issue.

**Comment by jkasten2** [comment]
@kistitan I decided to add a `null` guard here just in case there are any other possibles of this. Added this in commit b9f384d917303dab1142998d0a748c8ad871d054 and it will be included in the 3.4.4 release.
